src/utils/auth.ts

```diff
@@ -0,0 +1,40 @@
+import express, {Request, NextFunction} from 'express';
+const jwt = require('jsonwebtoken');
+import { PrismaClient } from '@prisma/client';
```

Comment:
I think you can import the User type from Prisma?
```suggestion
import { User, PrismaClient } from '@prisma/client';
```

Reply:
It was throwing a TS error when I tried to delete `req.user.password` while using the Prisma User because it wanted password to be optional, which it is not on the Prisma User type. I tried extending it and modifying the passward to make it optional, but that had issues too. So I just copied the Prisma type, but made password optional.

Reply:
ok that's fair

Reply:
Might be worth looking into how you're really supposed to handle deleting passwords when using Prisma, but for now this works. The only issue is it isn't extensible (eg. if new fields are added to User).

That being said, you don't necessarily have to request/attach the whole user object, you could just attach the ID and only load the user later if the route specifically requires it (often the ID is enough to check things like if a person is the owner of an object)